### PR TITLE
simwifi: Add the netlink event of connect/disconnect.

### PR DESCRIPTION
**Pull request by liqinhuixm**
## Summary
Add the netlink event of connect/disconnect for the simwifi function.
## Impact
N/A
## Testing
The app can receive the corresponding netlink event when the wifi is  connected successfully or disconnected.

### CODE REVIEW

File: arch/sim/src/sim/sim_wifidriver.c [hunk 1]
```diff
@@ -176,6 +178,23 @@ struct sim_netdev_s
   uint8_t network_id;           /* for sta, default is 0 */
 };
 
+/* for wireless event */
+
+struct wireless_event_s
+{
+  struct nlmsghdr  hdr;                 /* netlink message header */
```
Comment (pkarashchenko) [inline arch/sim/src/sim/sim_wifidriver.c:185]: Can we use same style as in above structure definitions?

File: arch/sim/src/sim/sim_wifidriver.c [hunk 2]
```diff
@@ -964,6 +1000,59 @@ static int freq_to_channel(uint16_t freq)
   return OK;
 }
 
+static int wifi_send_event(FAR struct net_driver_s *dev,
+                           unsigned int cmd, FAR union iwreq_data *wrqu)
+{
+  FAR struct wireless_event_list_s *alloc;
```
Comment (pkarashchenko) [inline arch/sim/src/sim/sim_wifidriver.c:1006]: Please remove FAR for sim

File: arch/sim/src/sim/sim_wifidriver.c [hunk 3]
```diff
@@ -964,6 +1000,59 @@ static int freq_to_channel(uint16_t freq)
...
+  FAR struct wireless_event_s *wev;
+
+  DEBUGASSERT(dev != NULL);
+
+  int up = IFF_IS_UP(dev->d_flags);
+
+  /* Allocate the response buffer */
+
+  alloc = (FAR struct wireless_event_list_s *)
```
Comment (pkarashchenko) [inline arch/sim/src/sim/sim_wifidriver.c:1015]: Same

File: arch/sim/src/sim/sim_wifidriver.c [hunk 4]
```diff
@@ -964,6 +1000,59 @@ static int freq_to_channel(uint16_t freq)
...
+  wev->attrevent.rta_len  = RTA_SPACE(sizeof(struct iw_event));
+  wev->attrevent.rta_type = IFLA_WIRELESS;
+  wev->event.len          = sizeof(union iwreq_data);
+  wev->event.cmd          = cmd;
+
+  memset(&wev->event.u, 0, sizeof(union iwreq_data));
+  memcpy(&wev->event.u, ((FAR char *)wrqu), sizeof(union iwreq_data));
+
+  netlink_add_broadcast(RTNLGRP_LINK,
+                        (FAR struct netlink_response_s *)alloc);
```
Comment (pkarashchenko) [inline arch/sim/src/sim/sim_wifidriver.c:1052]: Same

File: arch/sim/src/sim/sim_wifidriver.c (hunk 3 above)
Comment (liqinhuixm) [reply arch/sim/src/sim/sim_wifidriver.c]: Done

File: arch/sim/src/sim/sim_wifidriver.c (hunk 4 above)
Comment (liqinhuixm) [reply arch/sim/src/sim/sim_wifidriver.c]: Done

File: arch/sim/src/sim/sim_wifidriver.c (hunk 2 above)
Comment (liqinhuixm) [reply arch/sim/src/sim/sim_wifidriver.c]: Done

File: arch/sim/src/sim/sim_wifidriver.c (hunk 1 above)
Comment (liqinhuixm) [reply arch/sim/src/sim/sim_wifidriver.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**